### PR TITLE
Tests failing due to Motd #3932

### LINKED ISSUE #3932: Tests failing due to motd
These tests are failing. Note that we cannot expect to have a specific value for the motd in `build.properties` during testing because I use the same `build.properties` file for deploying to the live server.
AdminHomePageUiTest
FeedbackRubricQuestionUiTest
StudentFeedbackSubmitPageUiTest

### PR DESCRIPTION
**Pull request by taniach**
Fixes #3932

### CODE REVIEW

**Comment by taniach** [comment]
Dev green with empty Motd URL in build.properties

File: src/test/java/teammates/test/driver/HtmlHelper.java [hunk 1]
```diff
@@ -219,7 +221,29 @@ private static boolean isToolTip(Node currentNode) {
...
+            }
+                
+            for (int i = 0; i < attributes.getLength(); i++) {
+                Node attribute = attributes.item(i);
+                if (attribute.getNodeName().equalsIgnoreCase("id")
+                      && attribute.getNodeValue().contains("student-motd-container")) {
+                    return true;
+                }
+            }
+        }
```
Comment (damithc) [inline src/test/java/teammates/test/driver/HtmlHelper.java]: This for loop is going to run for every div right? Is there a faster way to determine this? Anyway, you can try it first. If it doesn't slow down tests, then no need to bother about it.

Comment (junhaoyap) [inline src/test/java/teammates/test/driver/HtmlHelper.java]: I don't think it's going to be a problem, isToolTip() does the same thing and checks every Node

File: src/test/java/teammates/test/driver/HtmlHelper.java [hunk 2]
```diff
@@ -219,7 +221,29 @@ private static boolean isToolTip(Node currentNode) {
...
+            return currentNode.getTextContent().contains("closeMotd");
+            
+        } else if (currentNode.getNodeName().equalsIgnoreCase("div")) {
+            NamedNodeMap attributes = currentNode.getAttributes();
+            
+            if (attributes == null) { 
+                return false;
+            }
+                
+            for (int i = 0; i < attributes.getLength(); i++) {
```
Comment (junhaoyap) [inline src/test/java/teammates/test/driver/HtmlHelper.java]: Would a for (Node attribute : attributes) { loop be better here?

**Comment by junhaoyap** [comment]
Looks ok other than the loop thing (which we can discuss and address), maybe we can single out a test with lots of HTML nodes and see whether adding these checks slows down the test by a significant margin?

File: src/test/java/teammates/test/driver/HtmlHelper.java (hunk 2 above)
Comment (taniach) [inline src/test/java/teammates/test/driver/HtmlHelper.java]: It doesn't work. Can only iterate over an array or an instance of java.lang.Iterable

Comment (junhaoyap) [inline src/test/java/teammates/test/driver/HtmlHelper.java]: Right, I digged further and found some resources on this:

http://stackoverflow.com/questions/4171380/generic-foreach-iteration-of-namednodemap

Apparently the name contains "map" but does not implements java.lang.iterable so we can't do that.

**Comment by junhaoyap** [comment]
Alright, let's merge this then.

**Comment by junhaoyap** [comment]
I'll run InstructorFeedbackResultsPage test on master vs on this branch to make sure the test doesn't slow down by much, it shouldn't but let's just make sure.

**Comment by junhaoyap** [comment]
I ran once on master and once on this branch, master took 190k ms, this branch took 198k ms so not that much to be worried about.

I also ran the entire test suite on this branch, remember you got stuck testing it just now? - I managed to get through the whole test suite so I think that is a one-off issue :)

**Comment by taniach** [comment]
Thanks for reviewing :)

**Pull request merged**